### DISCUSSION TITLE
Process for replacing deprecated formula

### DISCUSSION THREAD

**Discussion by jasonkarns** [Homebrew/discussions#4651] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.28-72-gbd5bd84
ORIGIN: https://github.com/Homebrew/brew
HEAD: bd5bd8437f96bce1576fbc46d64203f41fd848c7
Last commit: 11 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 062882103566d1af5d7a8aa9324e6d586a44cb46
Core tap last commit: 5 months ago
Core tap branch: master
Core tap JSON: 16 Jul 16:17 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CACHE: /Users/jasonkarns/.cache/Homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_NO_ANALYTICS: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.41.0 => /opt/homebrew/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  rake-completion
```

### Description of issue

Is there any existing process for replacing a deprecated formula? I understand there is a defined process for deprecating, disabling, and removal of formulae. And homebrew of course has two mechanisms for versioning formulae ( `@N` versioned formulae and revisions). Is there prior art for taking a deprecated formula (due to upstream being archived), and replacing it with its spiritual successor?

My guess would be to `@1` version the existing, deprecated formula. And replacing the unversioned formula with its replacement? (I'm not bothering with specifics here in this discussion, since I would presume that to be a discussion on the PR itself.)

**Comment by SMillerDev** [comment]
I'd just name the new formula differently to avoid confusing users.
